File: backend/src/main/java/com/shoppilot/entity/Product.java
package com.shoppilot.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("product")
public class Product {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long categoryId;

    private String name;

    private BigDecimal price;

    private Integer stock;

    private Integer status;

    private String description;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}

File: frontend/src/types/product.ts
export interface CategoryOption {
  id: number
  name: string
}

export interface Product {
  id: number
  categoryId: number
  categoryName: string
  name: string
  price: number
  stock: number
  status: number
  description?: string
  createdAt: string
  updatedAt: string
}

export interface ProductQuery {
  page: number
  size: number
  keyword?: string
  categoryId?: number
  status?: number
}

export interface ProductForm {
  categoryId?: number
  name: string
  price: number
  stock: number
  status: number
  description?: string
}

export interface PageResult<T> {
  records: T[]
  total: number
  page: number
  size: number
}
